### PR TITLE
Add initial bash completion support

### PR DESCRIPTION
**Pull request by arthurzam**
- Use flit-3.7 feature of 'external-data'
- add initial bash completion support

### CODE REVIEW

File: data/share/bash-completion/completions/nattka [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+			return
+			;;
+		--portage-conf | --repo)
+			COMPREPLY=($(compgen -d -- "${cur}"))
+			return
+			;;
+	esac
+
+	# find the subcommand
+	for ((i = 1; i < "${COMP_CWORD}"; i++)); do
```
Comment (mgorny) [inline data/share/bash-completion/completions/nattka:49]: I think quoting is unnecessary (and weird) here as well.

File: data/share/bash-completion/completions/nattka [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		case "${COMP_WORDS[i]}" in
+			--log-file | --api-key | --bugzilla-endpoint | --portage-conf | --repo)
+				((i++))
+				;;
+			-*) ;;
+			*)
+				cmd="${COMP_WORDS[i]}"
```
Comment (mgorny) [inline data/share/bash-completion/completions/nattka:56]:
```suggestion
				cmd=${COMP_WORDS[i]}
```

Plain assignments also don't need quoting.

**Review by mgorny — COMMENTED** [review]
Thanks for doing this but I feel a bit like this is a lot of extra work to maintain. Perhaps we should just use `dev-python/argcomplete` instead?

**Comment by arthurzam** [comment]
> Thanks for doing this but I feel a bit like this is a lot of extra work to maintain. Perhaps we should just use `dev-python/argcomplete` instead?

First of all, I'm all in maintaining the completions files myself. Either way, this is optional, meaning if a new feature is added without this matching completion, it is OK.

I have tried to use argcomplete (didn't know this exists). Added a very small change to `__main__.py`, and tried to use it. First of all there is a delay after each TAB, which is logical as it needs to run python - but this small delay we can ignore in the name of maintenance (maybe).
But I saw the completion was less refined and more error prone. Mine isn't full and complete, but it doesn't help where it can't, while argcomplete tries and does more spam in my opinion.

To summarize, it is of course your call how to go with it. I can open another PR using argcomplete and small bash completion file (doing just `eval "$(register-python-argcomplete nattka)"`).

File: data/share/bash-completion/completions/nattka [hunk 3]
```diff
@@ -0,0 +1,171 @@
+# bash completion for nattka
+
+_nattka() {
+	local i cmd cur prev words cword split
+	_init_completion || return
+
+	local subcommands="
```
[the remark this answers is not in the document]
Comment (mgorny) [reply data/share/bash-completion/completions/nattka:7]: No, `[@]` expands into separate args, `[*]` expands into a single IFS-separated string.

[the remark this answers is not in the document]
Comment (arthurzam) [reply data/share/bash-completion/completions/nattka]: Done and locally tested - thanks for fixing my misunderstanding.

**Review by mgorny — APPROVED** [review]
LGTM
